cmd/pmm-managed/main.go

```diff
@@ -237,6 +240,48 @@ func runDebugServer(ctx context.Context) {
 	cancel()
 }
 
+func runTelemetryService(ctx context.Context) {
+	l := logrus.WithField("component", "TELEMETRY")
+
+	disabled := strings.ToLower(os.Getenv("DISABLE_TELEMETRY"))
+	if disabled == "true" || disabled == "1" {
+		l.Infof("Telemetry is disabled by DISABLE_TELEMETRY env var")
+		return
+	}
+
+	cfg, _ := telemetry.LoadConfig(*telemetryConfigF)
+
+	updated, err := telemetry.CheckConfig(cfg)
+	if err != nil {
+		l.Warnf("cannot process telemetry condfig: %s", err)
+		l.Warnf("Telemetry is disabled")
+		return
+	}
+	if updated {
+		telemetry.SaveConfig(*telemetryConfigF, cfg)
+	}
+
+	// Using this env var for compatibility with the Toolkit
+	if telemetryEnvURL := os.Getenv("PERCONA_VERSION_CHECK_URL"); telemetryEnvURL != "" {
+		l.Infof("PERCONA_VERSION_CHECK_URL env var is set")
+		l.Infof("Using %s as the telemetry endpoint", telemetryEnvURL)
+		cfg.URL = telemetryEnvURL
+	}
+
+	svc, err := telemetry.NewService(cfg)
+	if err != nil {
+		l.Warnf("Cannot start telemetry: %s", err.Error())
+		return
+	}
+
+	l.Infof("Telemetry is enabled. Send data interval = %s seconds", cfg.Interval)
```

Comment:
cfg.Interval is `int`, so `%s` will give you `%!s(int=100500)`. I would prefer to define `Interval` as time.Duration. yaml package should be able to work with it. Alternatively, change `%s` to `%d`.